test/models/concerns/foreman/observable_model_test.rb

```diff
@@ -30,13 +30,23 @@ def self.name
         'model_info_updated.event.foreman',
       ]
 
-      assert_same_elements expected_event_subscription_hooks, klass.event_subscription_hooks
+      assert_same_elements expected_event_subscription_hooks, model_class.event_subscription_hooks
+    end
+
+    test 'notify with event context' do
+      ActiveSupport::Notifications.subscribed(callback, 'model_updated.event.foreman') do
+        callback.expects(:call).with do |_name, _started, _finished, _unique_id, payload|
+          payload[:context] == event_context
+        end
+
+        model.update(name: 'New Name')
+      end
     end
 
     test 'notify with default payload' do
       ActiveSupport::Notifications.subscribed(callback, 'model_updated.event.foreman') do
         callback.expects(:call).with do |_name, _started, _finished, _unique_id, payload|
-          payload == { object: model }.merge(event_context)
+          payload[:object] == model
```

Comment:
And I had to fix all of these comparsions, you cannot compare hash with indefferent access and regular hash easily.